src/modules/aws_iam/mappers/access_key.ts

```diff
@@ -0,0 +1,148 @@
+import isEqual from 'lodash.isequal';
+
+import {
+  AccessKeyMetadata,
+  IAM,
+  paginateListAccessKeys,
+  UpdateAccessKeyCommandInput,
+} from '@aws-sdk/client-iam';
+
+import { awsIamModule, AwsIamModule } from '..';
+import { AWS, crudBuilder2, paginateBuilder } from '../../../services/aws_macros';
+import { Context, Crud2, MapperBase } from '../../interfaces';
+import { AccessKey, accessKeyStatusEnum, IamUser } from '../entity';
+
+export class AccessKeyMapper extends MapperBase<AccessKey> {
+  module: AwsIamModule;
+  entity = AccessKey;
+  equals = (a: AccessKey, b: AccessKey) =>
+    isEqual(a.createDate, b.createDate) && Object.is(a.status, b.status);
+
+  async accessKeyMapper(e: AccessKeyMetadata, ctx: Context) {
+    const out = new AccessKey();
+    if (!e.AccessKeyId) return undefined;
+
+    out.accessKeyId = e.AccessKeyId;
+    if (e.CreateDate) out.createDate = e.CreateDate;
+    if (e.Status) out.status = e.Status as accessKeyStatusEnum;
+
+    if (e.UserName) {
+      // retrieve user and map it
+      if (!Object.values(ctx.memo?.cloud?.IamUser ?? {}).length) {
+        out.user =
+          (await awsIamModule.user.db.read(ctx, e.UserName)) ??
+          (await awsIamModule.user.cloud.read(ctx, e.UserName));
+      } else {
+        out.user =
+          (await awsIamModule.user.db.read(ctx, e.UserName)) ?? ctx?.memo?.cloud?.IamUser?.[e.UserName ?? ''];
+      }
+    }
+    return out;
+  }
+
+  getUserKeys = paginateBuilder<IAM>(
+    paginateListAccessKeys,
+    'AccessKeyMetadata',
+    undefined,
+    undefined,
+    UserName => ({ UserName }),
+  );
+
+  getAllKeys = paginateBuilder<IAM>(paginateListAccessKeys, 'AccessKeyMetadata');
+
+  async deleteAccessKey(client: IAM, userName: string, keyId: string) {
+    await client.deleteAccessKey({ UserName: userName, AccessKeyId: keyId });
+  }
+
+  updateAccessKey = crudBuilder2<IAM, 'updateAccessKey'>('updateAccessKey', input => input);
+
+  cloud = new Crud2<AccessKey>({
+    create: async (es: AccessKey[], ctx: Context) => {
+      // Do not cloud create, just restore database
+      await this.module.accessKey.db.delete(es, ctx);
+    },
+    read: async (ctx: Context, id?: string) => {
+      const client = (await ctx.getAwsClient()) as AWS;
+
+      if (id) {
+        const { keyId, user } = this.idFields(id);
```

Comment:
I do not see `user` decorated with `@cloudId`, how is this working?

Reply:
yep, an accidental removal... i will readd